using-mobile-extensions/adobe-analytics-edge/product-variable.md

```diff
@@ -1,6 +1,123 @@
+
+For more information on the products variable, see [Implementing a Merchandising Variable](https://experienceleague.adobe.com/docs/analytics/implementation/vars/plugins/addproductevar.html#vars).
+
+{% tabs %}
+{% tab title="Android" %}
+
+#### Java <a id="java-2"></a>
+
+#### Example <a id="example"></a>
+
+```java
+// build a product map for each product
+HashMap<String, Object> product1 = new HashMap<>();
+product1.put("SKU", "SH100");
+product1.put("name", "Running Shoes");
+product1.put("priceTotal", 69.95);
+product1.put("quantity", 1);
+HashMap<String, Object> product2 = new HashMap<>();
+product2.put("SKU", "SO100");
+product2.put("name", "Running Socks");
+product2.put("priceTotal", 29.99);
+product2.put("quantity", 10);
+
+// build a product list and add the product maps
+List<Object> productListItems = new ArrayList<>();
+productListItems.add(product1);
+productListItems.add(product2);
+
+// create an order total variable
+double orderTotal = 0D;
+orderTotal += (double)product1.get("priceTotal");
+orderTotal += (double)product2.get("priceTotal");
+
+// create a paymentsItem map which details the method of payment
+HashMap<String, Object> paymentsItem = new HashMap<>();
+paymentsItem.put("transactionID", "abc123");
+paymentsItem.put("paymentAmount", orderTotal);
+paymentsItem.put("paymentType", "credit_card");
+paymentsItem.put("currencyCode", "USD");
+
+// build an order, purchases, and commerce map
+HashMap<String, Object> order = new HashMap<>();
+order.put("purchaseID", "1234567890");
+order.put("priceTotal", orderTotal);
+order.put("currencyCode", "USD");
+order.put("payments", paymentsItem);
+
+HashMap<String, Object> purchases = new HashMap<>();
+purchases.put("value", 1.0);
+
+HashMap<String, Object> commerce = new HashMap<>();
+commerce.put("order", order);
+commerce.put("purchases", purchases);
+
+// create an Experience Event and send it using the AEP Edge extension
+HashMap<String, Object> xdmData = new HashMap<>();
+xdmData.put("commerce", commerce);
+xdmData.put("productListItems", productListItems);
+xdmData.put("eventType", "commerce.purchases");
+final ExperienceEvent experienceEvent = new ExperienceEvent.Builder().setXdmSchema(xdmData,"your-dataset-id").build();
+Edge.sendEvent(experienceEvent, new EdgeCallback() {
+  @Override
+  public void onComplete(List<EdgeEventHandle> list) {
+    // handle the response
+  }
+});
+}
+```
+
+The alternative to using raw dictionaries for your XDM data is building a concrete class based on your XDM schema. For more information, see the Schema interface at https://aep-sdks.gitbook.io/docs/v/AEP-Edge-Docs/using-mobile-extensions/adobe-edge/edge-api-reference#public-classes, and the commerce example in [AEP SDK Sample App for Android](https://github.com/adobe/aepsdk-sample-app-android).
```

Comment:
.. see the Schema interface in Edge API reference Public classes <link> and the ...